bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/handler/BoschAccountHandler.java

```diff
@@ -61,7 +61,7 @@ public BoschAccountHandler(Bridge bridge, HttpClient httpClient, OAuthFactory oA
 
         this.oAuthFactory = oAuthFactory;
 
-        oAuthClientService = oAuthFactory.createOAuthClientService(getThing().getUID().getAsString(), BSK_TOKEN_URI,
+        oAuthClientService = oAuthFactory.createOAuthClientService(thing.getUID().getAsString(), BSK_TOKEN_URI,
```

Comment:
You need to move the creation of oAuthClientService in `initialize`.
If not, this will not work in case the thing handler is initialized, disposed and then again initialized.

I did that for another binding.

Reply:
Nice catch, thanks! Fixed.

Reply:
More refactoring is needed since the object is exposed to the child handler and injected into the API controller.

Reply:
Done.